integrity: InvertedIndex unclear which file broken

```
 go run ./cmd/erigon seg integrity --datadir /erigon-data/ --check=InvertedIndex
INFO[07-05|04:25:24.543] logging to file system                   log dir=/home/ubuntu/.local/share/erigon/logs file prefix=erigon log level=info json=false
INFO[07-05|04:25:24.544] logging to file system                   log dir=/erigon-data/logs file prefix=erigon log level=info json=false
INFO[07-05|04:25:24.545] [db] open                                label=chaindata sizeLimit=1TB pageSize=4KB
INFO[07-05|04:25:28.244] [snapshots:block] Stat                   blocks=73.59M indices=73.59M alloc=0B sys=0B
INFO[07-05|04:25:28.593] [snapshots:bor] Stat                     blocks=73.59M indices=73.59M alloc=0B sys=0B
INFO[07-05|04:25:51.293] [snapshots:history] Stat                 blocks=73.58M txNum2blockNum="2048=49361K,3072=65048K,3328=69764K,3456=72384K,3520=73534K,3522=73570K,3523=73588K" txs=5.50B first_history_idx_in_db=73588953 alloc=0B sys=0B
INFO[07-05|04:25:51.293] [downloads]                              locked=true at="05 Jul 25 03:10 2025"
panic: runtime error: slice bounds out of range [45603844:45603843]

goroutine 2658 [running]:
github.com/erigontech/erigon-lib/seg.(*Getter).NextUncompressed(0xc0f14d2c98)
	/home/ubuntu/erigon/erigon-lib/seg/decompress.go:752 +0xf2
github.com/erigontech/erigon-lib/state.(*InvertedIndexRoTx).IntegrityInvertedIndexAllValuesAreInRange.func1({0x0, 0x0, 0x0, 0x5f5e100, 0x0, 0xc0c3b09b40})
	/home/ubuntu/erigon/erigon-lib/state/integrity.go:153 +0x22a
github.com/erigontech/erigon-lib/state.(*InvertedIndexRoTx).IntegrityInvertedIndexAllValuesAreInRange(0xc000bd04d0, {0x36efa00, 0x61fa8a0}, 0x1, 0x0)
	/home/ubuntu/erigon/erigon-lib/state/integrity.go:193 +0x1a3
github.com/erigontech/erigon-lib/state.(*AggregatorRoTx).IntegrityInvertedIndexAllValuesAreInRange(0xc0e9246a80, {0x36efa00, 0x61fa8a0}, {0x2e40bb0, 0x11}, 0x1, 0x0)
	/home/ubuntu/erigon/erigon-lib/state/integrity.go:57 +0x105
github.com/erigontech/erigon/eth/integrity.E3EfFiles.func1()
	/home/ubuntu/erigon/eth/integrity/e3_ef_files.go:43 +0x170
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/ubuntu/go/pkg/mod/golang.org/x/sync@v0.10.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
	/home/ubuntu/go/pkg/mod/golang.org/x/sync@v0.10.0/errgroup/errgroup.go:75 +0x93
exit status 2
```

--- a/erigon-lib/state/integrity.go
+++ b/erigon-lib/state/integrity.go
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"github.com/erigontech/erigon-lib/common/dbg"
 	"path/filepath"
 	"time"
 
@@ -146,7 +147,16 @@ func (iit *InvertedIndexRoTx) IntegrityInvertedIndexAllValuesAreInRange(ctx cont
 
 	logEvery := time.NewTicker(30 * time.Second)
 	defer logEvery.Stop()
-	iterStep := func(item visibleFile) error {
+	iterStep := func(item visibleFile) (err error) {
+		defer func() {
+			if r := recover(); r != nil {
+				fileName := "not found"
+				if item.src != nil && item.src.decompressor != nil {
+					fileName = item.src.decompressor.FileName()
+				}
+				err = fmt.Errorf("panic in file: %s\n%s", fileName, dbg.Stack())
+			}
+		}()
 		item.src.decompressor.MadvSequential()
 		defer item.src.decompressor.DisableReadAhead()
 

--- a/erigon-lib/state/integrity.go
+++ b/erigon-lib/state/integrity.go
@@ -154,7 +154,7 @@ func (iit *InvertedIndexRoTx) IntegrityInvertedIndexAllValuesAreInRange(ctx cont
 				if item.src != nil && item.src.decompressor != nil {
 					fileName = item.src.decompressor.FileName()
 				}
-				err = fmt.Errorf("panic in file: %s\n%s", fileName, dbg.Stack())
+				err = fmt.Errorf("panic in file: %s. Stack: %s", fileName, dbg.Stack())
 			}
 		}()
 		item.src.decompressor.MadvSequential()

